core/computed/metrics/time-to-first-byte.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license Copyright 2023 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {makeComputedArtifact} from '../computed-artifact.js';
+import {NavigationMetric} from './navigation-metric.js';
+import {MainResource} from '../main-resource.js';
+import {NetworkAnalysis} from '../network-analysis.js';
+
+class TimeToFirstByte extends NavigationMetric {
+  /**
+   * @param {LH.Artifacts.NavigationMetricComputationData} data
+   * @param {LH.Artifacts.ComputedContext} context
+   * @return {Promise<LH.Artifacts.Metric>}
+   */
+  static async computeSimulatedMetric(data, context) {
+    const mainResource = await MainResource.request(data, context);
+    const networkAnalysis = await NetworkAnalysis.request(data.devtoolsLog, context);
+
+    const observedTTFB = (await this.computeObservedMetric(data, context)).timing;
+    const observedResponseTime =
+      networkAnalysis.serverResponseTimeByOrigin.get(mainResource.parsedURL.securityOrigin);
+    if (!observedResponseTime) throw new Error('Could not compute response time for origin');
+
+    // Estimate when the connection is not warm.
+    // TTFB = DNS + (SSL)? + TCP handshake + 1 RT for request + server response time
+    let roundTrips = 3;
+    if (mainResource.parsedURL.scheme === 'https') roundTrips += 1;
+    const estimatedTTFB = data.settings.throttling.rttMs * roundTrips + observedResponseTime;
+
+    const timing = Math.max(observedTTFB, estimatedTTFB);
+    return {timing};
+  }
+
+  /**
+   * @param {LH.Artifacts.NavigationMetricComputationData} data
+   * @param {LH.Artifacts.ComputedContext} context
+   * @return {Promise<LH.Artifacts.Metric>}
+   */
+  static async computeObservedMetric(data, context) {
+    const {processedNavigation} = data;
+    const timeOriginTs = processedNavigation.timestamps.timeOrigin;
+    const mainResource = await MainResource.request(data, context);
+
+    const timestamp = mainResource.responseHeadersEndTime * 1000;
```

Comment:
TTFB is technically the _start_ of response headers (which AFAIK isn't available to us?). I wonder how much that will affect the values

Reply:
Yeah that worried me as well. I didn't think it was worth adding the signal to CDP though.

Reply:
can you add a comment about that, maybe in the fileoverview or just inline here? It's helpful for people coming in later figuring out how things work to see that a) called out without having to figure it out themselves and b) that it wasn't a mistake